File: 05964.cpp
#include <iostream>
#include <stack>
#define MOD 12345678910LL

using namespace std;

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int n;
	cin >> n;
	stack<long long> st;
	for(int i = 0; i < n; ++i){
		int x;
		cin >> x;
		if(x){
			long long sum = 0;
			while(!st.empty() && st.top() != 0){
				sum += st.top();
				sum %= MOD;
				st.pop();
			}
			st.pop();
			if(sum == 0){
				st.push(1);
			}else{
				st.push(2 * sum % MOD);
			}
		}else{
			st.push(0);
		}
	}
	long long ans = 0;
	while(!st.empty()){
		ans += st.top();
		ans %= MOD;
		st.pop();
	}
	cout << ans;
	return 0;
}

File: 05964.py
import sys


n = int(sys.stdin.readline())
stack = []
for _ in range(n):
    x = sys.stdin.readline().rstrip()
    if x == "1":
        entry = 0
        while stack and stack[-1] != 0:
            entry += stack.pop()
        stack.pop()
        if entry == 0:
            stack.append(1)
        else:
            stack.append(2 * entry)
    else:
        stack.append(0)
ans = 0
while stack:
    ans += stack.pop()
print(ans % 12345678910)
